### ISSUE TITLE
@$() doesn't split quoted strings correctly

### ISSUE THREAD

**Issue by asmeurer** [xonsh/xonsh#1091]
The docs claim that `@$(cmd)` does `@([i.strip() for i in $(cmd).split()])`, which is correct, but not helpful, because if `cmd` contains quoted strings with spaces, they are not parsed as you would hope. In general, there is no way to escape spaces with this.

**Comment by scopatz** [comment]
Yep, I agree. This is a first pass. We need something that splits on whitespace that is not in a string, by default. This should be a do-able regex.

**Comment by adqm** [comment]
FWIW, bash seems to do what is described in the docs:
```
$ ls `echo \"this dir\"`
ls: cannot access "this: No such file or directory
ls: cannot access dir": No such file or directory
```

That said, I think better handling of strings would be nice, since that's almost certainly what people intend.

**Comment by wreed4** [comment]
shlex.split does exactly what you need instead of the default str.split used above.

**Comment by scopatz** [comment]
I think this should be implemented with the lexer itself.

**Comment by scopatz** [comment]
Potential fix in #2227

**Comment by gforsyth** [comment]
Closing this out since #2227 was merged but please reopen if the issue persists.
